pycloudlib/ec2/instance.py

```diff
@@ -231,17 +246,21 @@ def _attach_network_interface(self, interface_id: str) -> str:
 
         response = self._client.attach_network_interface(**args)
 
-        self._instance.reload()
-
-        for nic in self._instance.network_interfaces:
-            if nic.attachment["AttachmentId"] == response["AttachmentId"]:
-                nic.modify_attribute(
-                    Attachment={
-                        "AttachmentId": response["AttachmentId"],
-                        "DeleteOnTermination": True,
-                    }
-                )
-                return nic.private_ip_address
+        # It's possible the attach worked correctly but it's not yet
+        # present in the nic attributes
+        for _ in range(5):
+            self._instance.reload()
+            for nic in self._instance.network_interfaces:
+                if nic.attachment["AttachmentId"] == response["AttachmentId"]:
+                    nic.modify_attribute(
+                        Attachment={
+                            "AttachmentId": response["AttachmentId"],
+                            "DeleteOnTermination": True,
```

Comment:
Doesn't  this [`DeleteOnTermination`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/modify_network_interface_attribute.html#EC2.Client.modify_network_interface_attribute) mean we don't have to worry about memoizing created_interfaces on AWS?

Reply:
You'd think so, but I kept seeing NICs sticking around. I put a comment about it in the `delete()` method.